src/flask/typing.py

```diff
@@ -0,0 +1,44 @@
+import typing as t
+
+
+if t.TYPE_CHECKING:
+    from werkzeug.datastructures import Headers  # noqa: F401
+    from .wrappers import Response  # noqa: F401
+
+# The possible types that are directly convertible or are a Response object.
+ResponseValue = t.Union[
+    "Response",
+    t.AnyStr,
+    t.Dict[str, t.Any],  # any jsonify-able dict
+    t.Generator[t.AnyStr, None, None],
+]
+StatusCode = int
+
+# the possible types for an individual HTTP header
+HeaderName = str
+HeaderValue = t.Union[str, t.List[str], t.Tuple[str, ...]]
+
+# the possible types for HTTP headers
+HeadersValue = t.Union[
+    "Headers", t.Dict[HeaderName, HeaderValue], t.List[t.Tuple[HeaderName, HeaderValue]]
+]
+
+# The possible types returned by a route function.
+ResponseReturnValue = t.Union[
```

Comment:
Technically this can also include `wsgiref.types.WSGIApplication`.

Reply:
Added.